src/base/bittorrent/session.cpp

```diff
@@ -1729,35 +1718,22 @@ bool Session::findIncompleteFiles(TorrentInfo &torrentInfo, QString &savePath) c
 {
     auto findInDir = [](const QString &dirPath, TorrentInfo &torrentInfo) -> bool
     {
+        const QDir dir(dirPath);
         bool found = false;
-        if (torrentInfo.filesCount() == 1) {
-            const QString filePath = dirPath + torrentInfo.filePath(0);
-            if (QFile(filePath).exists()) {
-                found = true;
-            }
-            else if (QFile(filePath + QB_EXT).exists()) {
+        for (int i = 0; i < torrentInfo.filesCount(); ++i) {
+            QString filePath = torrentInfo.filePath(i);
+            if (dir.exists(filePath)) {
                 found = true;
-                torrentInfo.renameFile(0, torrentInfo.filePath(0) + QB_EXT);
             }
-        }
-        else {
-            QSet<QString> allFiles;
-            int dirPathSize = dirPath.size();
-            foreach (const QString &file, findAllFiles(dirPath + torrentInfo.name()))
-                allFiles << file.mid(dirPathSize);
-            for (int i = 0; i < torrentInfo.filesCount(); ++i) {
-                QString filePath = torrentInfo.filePath(i);
-                if (allFiles.contains(filePath)) {
+            else {
```

Comment:
@fbriere, @Chocobo1 mean you should collapse subblock in this case: `else if (...)`.